src/System.Private.CoreLib/src/System/MulticastDelegate.cs

```diff
@@ -135,32 +136,35 @@ public override sealed bool Equals(object obj)
         private bool InvocationListEquals(MulticastDelegate d)
         {
             Debug.Assert(d != null && (_invocationList as object[]) != null, "bogus delegate in multicast list comparison");
```

Comment:
```
Debug.Assert(d != null);
Debug.Assert(_invocationList is object[]);
object[] invocationList = (object[])_invocationList;
```

This would make it consistent with other similar places in this file (https://github.com/dotnet/coreclr/pull/23615/files#diff-53dcb7b3c9c67908fe7c008e271cffaaR309) and avoid all the pixie-dust.

Reply:
Sure. For consistency, long message or not? I would rather not, since it's debug only, I probably have the ability to investigate directly (this isn't the scenario of a customer posting a stack trace)

`Debug.Assert(invocationList[i] is Delegate, $"Expected {typeof(Delegate)}, got {invocationList[i].GetType()}");`

Reply:
I do not think you need to log messages.

Reply:
OK